pkg/operator/controllers/base/aro_controller.go

```diff
@@ -13,16 +13,21 @@ import (
 	"k8s.io/apimachinery/pkg/types"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
 
 	arov1alpha1 "github.com/Azure/ARO-RP/pkg/operator/apis/aro.openshift.io/v1alpha1"
 )
 
 type AROReconciler interface {
+	reconcile.Reconciler
 	GetName() string
 	SetupWithManager(ctrl.Manager) error
+	ReconcileEnabled(context.Context, ctrl.Request, *arov1alpha1.Cluster) (ctrl.Result, error)
```

Comment:
nit: these names may convey that these functions return a boolean (e.g. `isReconcileEnabled`). I can't think of an immediately better name, but this function and `ReconcileDisabled` may need a better name.

Reply:
Best I can think of is flip-flop it to `EnabledReconcile` but I'm not sure if that's any clearer.
Naming is hard.  I'll submit to the consensus of opinion.

Reply:
Java uses `is` as a standard prefix :coffee: https://www.comp.nus.edu.sg/~cs2103/AY1516S2/contents/coding-standards-java.html - since that was my first language, it's my preference too. It is longer though, and we don't have a coding standard :bangbang: so for now I'd say stick with what feels right to you.

Reply:
I take that to mean the method name is clear enough as is then, if I'm understanding you right, because the expectation for a simple "is reconcile enabled?" test is that it would be named `IsReconcileEnabled`.  Whereas this method is what the controller implements for "reconcile when enabled flag is true". (Note also the companion "reconcile when enabled flag is false" method on the next line down).

We can always rename it later if it proves confusing to others, but I think in the proper context the confusion @tsatam was calling out should be avoided.

Reply:
I think that we should keep `Reconcile` as is and use `context.WithValue` instead because it seems to me that it fits this use case. In fact one could use `context.WithValue` to pass along whether or not the `Reconcile` function should be acting in an enabled/disabled mode too.

Imho this would simplify the implementation a bit and return control of how the enabled/disabled logic worked back to each reconcile function.